src/app/DomainPermissionsPage.qml

```diff
@@ -0,0 +1,290 @@
+                iconName: "select"
+                visible: selectMode
+                onTriggered: {
+                    if (domainPermissionsListView.ViewItems.selectedIndices.length === domainPermissionsListView.count) {
+                        domainPermissionsListView.ViewItems.selectedIndices = []
+                    } else {
+                        var indices = []
+                        for (var i = 0; i < domainPermissionsListView.count; ++i) {
+                            indices.push(i)
+                        }
+                        domainPermissionsListView.ViewItems.selectedIndices = indices
+                    }
+                }
+            },
+            Action {
+                text: i18n.tr("Delete")
+                iconName: "delete"
+                visible: selectMode
+                enabled: domainPermissionsListView.ViewItems.selectedIndices.length > 0
+                onTriggered: {
+                    var toDelete = []
+                    for (var index = 0; index < domainPermissionsListView.ViewItems.selectedIndices.length; index++) {
+                        var selectedDomainSetting = domainPermissionsListView.model.get(domainPermissionsListView.ViewItems.selectedIndices[index])
+                        toDelete.push(selectedDomainSetting.domain)
+                    }
+                    for (var i = 0; i < toDelete.length; i++) {
+                        DomainPermissionsModel.removeEntry(toDelete[i])
+                    }
+                    domainPermissionsListView.ViewItems.selectedIndices = []
+                    selectMode = false
+                }
+            },
+            Action {
+                iconName: "edit"
+                visible: !selectMode
+                enabled: domainPermissionsListView.count > 0
+                onTriggered: {
+                    selectMode = true
+                }
+            },
+            Action {
+                iconName: "add"
+                visible: !selectMode
+
+                onTriggered: {
+                    var promptDialog = PopupUtils.open(Qt.resolvedUrl("PromptDialog.qml"), domainPermissionsPage);
+                    promptDialog.title = i18n.tr("Add domain")
+                    promptDialog.message = i18n.tr("Enter the name of the domain, e.g. example.com (subdomains will be removed).")
+                    promptDialog.accept.connect(function(text) {
+                        if (text !== "") {
+                            var domain = DomainPermissionsModel.getDomainWithoutSubdomain(UrlUtils.extractHost(text));
+                            if (DomainPermissionsModel.contains(domain)) {
+                                domainPermissionsItem.setDomainAsCurrentItem(domain);
+                            }
+                            else {
+                                DomainPermissionsModel.insertEntry(domain, false);
+                                reload(domain);
+                            }
```

Comment:
Wait, what are we doing if the user enters nothing? Does `domain` exist here?

Reply:
there is the check `if (text !== "") {` in line 119

Reply:
Ah, okay.